vaadin-dropdown-menu.html

```diff
@@ -14,40 +64,142 @@
+          this.inputElement.addEventListener('keydown', e => this._onKeyDown(e));
+        }
+
+        _onKeyDown(e) {
+          if (!this.opened && /^(ArrowDown|Down|ArrowUp|Up|Enter|Space)$/.test(e.code)) {
+            e.preventDefault();
+            this._toggleOverlay();
+          }
+        }
+
+        _openedChanged(opened) {
+          if (opened && !this._instance) {
+            this._contentTemplate = this.querySelector('template');
+            const Templatizer = Polymer.Templatize.templatize(this._contentTemplate, this, {
+              instanceProps: {
+                detail: true,
+                target: true
+              },
+              forwardHostProp: function(prop, value) {
+                if (this._instance) {
+                  this._instance.forwardHostProp(prop, value);
+                }
+              }
+            });
+            this._instance = new Templatizer({});
+            this.overlayElement.content = this._instance.root;
+          }
+
+          this.overlayElement.opened = opened;
+
+          let menu;
+          const selectedChangedListener = (e) => {
+            const selectedItem = menu._navItems[e.detail.value];
+            this.value = selectedItem ? selectedItem.textContent : '';
+
+            this.opened = false;
+          };
+
+          if (opened) {
+            menu = this.$.overlay.$.content.lastElementChild;
+            if (!menu || !menu._navItems) {
+              return;
+            }
+
+            menu.addEventListener('selected-changed', selectedChangedListener);
+
+            const itemToFocus = this.value ? menu._navItems.filter(item => item.textContent === this.value)[0] : menu._navItems[0];
+            if (itemToFocus) {
+              itemToFocus.focus();
+            }
+          } else {
+            if (menu) {
+              menu.removeEventListener('selected-changed', selectedChangedListener);
+            }
+
+            this.inputElement.focus();
+          }
+        }
+
+        _toggleOverlay() {
+          this.opened = !this.opened;
+          const inputRect = this.inputElement.getBoundingClientRect();
+
+          this.overlayElement.style.left = inputRect.left + 'px';
+          this.overlayElement.style.top = (inputRect.top > window.innerHeight - inputRect.top) ?
+            this.overlayElement.style.top = inputRect.bottom - this.overlayElement.$.overlay.offsetHeight + 'px' : inputRect.top + 'px';
+        }
+
+        get overlayElement() {
+          return this.$.overlay;
+        }
+
+        get inputElement() {
+          return this.$.input;
+        }
       }
 
       customElements.define(VaadinDropdownMenuElement.is, VaadinDropdownMenuElement);
 
       /**
-       * @namespace Vaadin
-       */
+        * @namespace Vaadin
+        */
       window.Vaadin = window.Vaadin || {};
-      Vaadin.VaadinDropdownMenuElement = VaadinDropdownMenuElement;
+      Vaadin.DropdownMenuElement = VaadinDropdownMenuElement;
```

Comment:
As other vaadin elements, I think you should change `VaadinDropdownMenuElement ` to `DropdownMenuElement`.

Reply:
@manolo check:

`vaadin-tabs` & `vaadin-tab`:
https://github.com/vaadin/vaadin-tabs/blob/12bd28a6cd924a3caf3f9f4c97f4bc89bf878368/vaadin-tabs.html#L235-L236

https://github.com/vaadin/vaadin-tabs/blob/12bd28a6cd924a3caf3f9f4c97f4bc89bf878368/vaadin-tab.html#L102-L103

`vaadin-button`:
https://github.com/vaadin/vaadin-button/blob/a4972d26c8934fcfbf99b56ede420d76cc03a86a/vaadin-button.html#L108-L109

`vaadin-radio-button`:
https://github.com/vaadin/vaadin-radio-button/blob/4a250e6b8b2cada6e2338be1d143669860a8d7f6/vaadin-radio-button.html#L201-L202

Reply:
Dont worry, thanks Manolo!